extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -80,6 +98,33 @@ export class AmpAd3PImpl extends AMP.BaseElement {
 
     /** @private {?Promise} */
     this.layoutPromise_ = null;
+
+    // Carve-outs: We only want to enable profiling pingbacks when:
+    //   - The ad is from one of the Google networks (AdSense or Doubleclick).
+    //   - The ad slot is in the A4A-vs-3p amp-ad control branch (either via
+    //     internal, client-side selection or via external, Google Search
+    //     selection).
+    //   - We haven't turned off profiling via the rate controls in
+    //     build-system/global-config/{canary,prod}-config.json
+    // If any of those fail, we use the `NullLifecycleReporter`, which is a
+    // a no-op (sends no pings).
+    const type = element.getAttribute('type');
+    const eid = element.getAttribute(EXPERIMENT_ATTRIBUTE);
+    const isGoogleControlBranch =
+        (eid == ADSENSE_A4A_EXTERNAL_EXPERIMENT_BRANCHES.control) ||
+        (eid == ADSENSE_A4A_INTERNAL_EXPERIMENT_BRANCHES.control) ||
+        (eid == DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES.control) ||
+        (eid == DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES.control);
+    randomlySelectUnsetPageExperiments(this.win, PROFILING_RATE);
+    if ((type == 'doubleclick' || type == 'adsense') &&
+        isGoogleControlBranch &&
+        isExperimentOn(this.win, 'a4aProfilingRate')) {
+      this.lifecycleReporter_ =
+          new AmpAdLifecycleReporter(this.win, this.element, 'amp');
+    } else {
+      this.lifecycleReporter_ = new NullLifecycleReporter();
+    }
+    this.lifecycleReporter_.sendPing('constructor');
```

Comment:
maybe call it something like `adSlotBuilt`, which could have a counterpart in 3p as well? `constructor` is a too vague and generic term :-)